### PR TITLE
Player service

### LINKED ISSUE #58: Player service
The player service is responsibly for general player management. It should support managing player data in the database, fetching of offline and online players, deleting player info etc.

The player service is used both in the core and by modules to do various things on players.

### PR DESCRIPTION
**Pull request by snixtho**
closes #58 

- Add player service.
- Add player parser for command args.
- Add player flags parsing.
- Add basic mod commands for players.

### CODE REVIEW

File: src/Modules/Player/ModPermissions.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace EvoSC.Modules.Official.Player;
+
+[PermissionGroup]
+public enum ModPermissions
+{
+    [Description("Can kick players from the server.")]
+    KickPlayer,
+    
+    [Description("Can b or blacklist players from the server.")]
```
Comment (AtomicLiquid) [inline src/Modules/Player/ModPermissions.cs:12]:
```suggestion
    [Description("Can ban or blacklist players from the server.")]
```

File: src/Modules/Player/Services/PlayerService.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            await _playerManager.UpdateLastVisitAsync(player);
+            await _server.SendChatMessage($"$<{player.NickName}$> joined!");
+        }
+    }
+
+    public async Task KickAsync(IPlayer player, IPlayer actor)
+    {
+        if (await _server.Remote.KickAsync(player.GetLogin()))
+        {
+            await _server.SendChatMessage($"$284{player.NickName} was kicked.", actor);
```
Comment (AtomicLiquid) [inline src/Modules/Player/Services/PlayerService.cs:43]: I think we have to create some sort of config to adjust the colors here, so they're not hardcoded. Possibly create an issue?

**Review by AtomicLiquid — DISMISSED** [review]
Good stuff, make sure to take a look at the warnings from Github Actions.

File: src/EvoSC.Commands/CommandInteractionHandler.cs [hunk 3]
```diff
@@ -70,10 +71,15 @@ async Task HandleUserErrors(IParserResult result, string playerLogin)
...
             }
-            
+
             var message = $"Error: {result.Exception.Message}";
             await _serverClient.SendChatMessage($"Error: {message}", playerLogin);
         }
+
+        if (result.Exception is PlayerNotFoundException playerNotFoundException)
+        {
+            await _serverClient.SendChatMessage($"Error: {playerNotFoundException.Message}", playerLogin);
```
[the remark this answers is not in the document]
Comment (snixtho) [reply src/EvoSC.Commands/CommandInteractionHandler.cs:81]: Yep, that particular message is made user friendly.

File: src/Modules/Player/Services/PlayerService.cs (hunk 2 above)
Comment (snixtho) [reply src/Modules/Player/Services/PlayerService.cs:43]: Yes, the issue was already made: https://github.com/EvoTM/EvoSC-sharp/issues/68

**Comment by snixtho** [comment]
> Good stuff, make sure to take a look at the warnings from Github Actions.

Fixes some warnings. Don't think the rest of the warnings need to/should be fixed as they are related to how the database models work.

**Pull request merged**